### DISCUSSION TITLE
Bug in InChI parsing?

### DISCUSSION THREAD

**Discussion by simonmb** [rdkit/rdkit#4600] [Q&A] [answered]
I have the following InChI taken from Pubchem: 
`InChI=1S/C2H5NS/c1-2(3)4/h1H3,(H2,3,4)`

it is supposed to be for thioacetamide with the following SMILES:
`CC(=S)N`

However when running
`Chem.MolToSmiles(Chem.MolFromInchi('InChI=1S/C2H5NS/c1-2(3)4/h1H3,(H2,3,4)'))`

it returns:
`CC(=N)S`

Either there is an error in the InChI parser or the InChI is wrong. But Pubchem and the official CAS registry give the same InChI.

When I run the test backwards:
`Chem.MolToInchi(Chem.MolFromSmiles('CC(=S)N'))`

it returns the InChI from Pubchem:
`InChI=1S/C2H5NS/c1-2(3)4/h1H3,(H2,3,4)`

So it seems like a bug, or am I missing something?

**Comment by greglandrum** [comment, marked as answer]
The InChI algorithm does tautomer normalization, so the InChIs for `CC(=N)S` and `CC(=S)N` are the same:
```
>>> Chem.MolToInchi(Chem.MolFromSmiles('CC(=N)S'))
'InChI=1S/C2H5NS/c1-2(3)4/h1H3,(H2,3,4)'
>>> Chem.MolToInchi(Chem.MolFromSmiles('CC(=S)N'))
'InChI=1S/C2H5NS/c1-2(3)4/h1H3,(H2,3,4)'
```

**Reply by simonmb** [reply]
Is there a way to turn the tautomer normalization off?

**Reply by greglandrum** [reply]
There is, but a big part of the idea of InChI is that it does tautomer normalization. What are you trying to do?
FYI: If you want to store molecules in a canonical, but non-tautomer normalized form, you can just use SMILES.

**Reply by simonmb** [reply]
Thank you, this was the information I needed. I am using the SMILES instead then.
